packages/rule-sri/README.md

```diff
@@ -0,0 +1,145 @@
+# Require scripts and styles to use Subresource integrity (`sri`)
+
+`sri` warns about requesting scripts or styles without using Subresource
+integrity.
+
+## Why is this important?
+
+Nowadays it's very common to use third party resources from CDNs or different
+services (analytics, ads, etc.) and thus increasing the risk surface of your
+web application.
+
+While there are techniques to verify the agent is talking with the right server
+(TLS, HSTS, etc.), an attacker (or administrator) with access to the server can
+manipulate the content with impunity.
+
+> If you want to load a crypto miner on 1,000+ websites you don't attack 1,000+
+websites, you attack the 1 website that they all load content from.
+([Scott Helme][weak link])
+
+Subresource integrity [is a standard][sri spec] that mitigates this by ensuring
+that an exact representation of a resource, and only that representation, loads
+and executes.
+
+## What does the rule check?
+
+This rule checks that a website uses correctly SRI, more especifically:
+
+* All the downloaded resources by an `<script>` or `<link rel="stylesheet">`
+  have an `integrity` attribute.
+* [The `integrity` attribute has to be valid][sri format]. I.e.: it should
+  contain something in the form of `sha(256|384|512)-HASH`, where `HASH` is
+  the hashed value of the downlaoded body's response using the previous
+  algorithm (`sha256`, `sha384`, or `sha512`).
+* The minium cryptographic hash function used has to be [`sha384`][collisions].
+  If multiple are provided, the highest one will be used to determine if the
+  baseline is met.
+* When using a cross-origin resource (e.g.: using a script hosted in a third
+  party CDN), the `<script>` tag needs to have a valid
+  [`crossorigin` attribute][crossorigin].
+* The resource needs to be served on a [secure context][secure context]
+  (i.e.: HTTPS)
+* The hash from the `integrity` attribute needs to be the same as the one
+  calculated using the response's body.
+* If multiple hashes are provided, at least one needs to be valid
+
+### Examples that **trigger** the rule
+
+Same-origin resource with no `integrity`:
+
+```html
+<link rel="stylesheet" href="/style.css">
+```
+
+Same-origin resource with hash function less secure than `sha384`:
+
+```html
+<script src="/script.js" integrity="sha256-validHashHere">
```

Comment:
Maybe to be consistent we should use a random hash?

Reply:
I was thinking about that, but I though it was clearer this way so users reading about SRI for the first time know they need to calculate the hash and replace it.

Reply:
ok.